CMakeLists.txt

```diff
@@ -119,26 +119,25 @@ if(APPLE)
   if(IS_DIRECTORY "/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.10.sdk")
     set(HAVE_10_10_SDK TRUE)
   endif()
-
   # Try for the lowest version SDK we can
-  if(HAVE_10_8_SDK)
-      set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -headerpad_max_install_names")
-      set(CMAKE_OSX_DEPLOYMENT_TARGET 10.8 CACHE STRING "" FORCE)
-      set(CMAKE_OSX_SYSROOT "/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.8.sdk" CACHE PATH "" FORCE)
-  elseif(HAVE_10_9_SDK)
-      set(CMAKE_C_COMPILER clang)
-      set(CMAKE_CXX_COMPILER clang++)
-      set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -headerpad_max_install_names")
-      set(CMAKE_OSX_DEPLOYMENT_TARGET 10.9 CACHE STRING "" FORCE)
-      set(CMAKE_OSX_SYSROOT "/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.9.sdk" CACHE PATH "" FORCE)
-  elseif(HAVE_10_10_SDK)
-      set(CMAKE_C_COMPILER clang)
-      set(CMAKE_CXX_COMPILER clang++)
-      set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -headerpad_max_install_names")
-      set(CMAKE_OSX_DEPLOYMENT_TARGET 10.10 CACHE STRING "" FORCE)
-      set(CMAKE_OSX_SYSROOT "/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.10.sdk" CACHE PATH "" FORCE)
+  if ((${OSX_VER_MAJOR} VERSION_EQUAL 8) AND (HAVE_10_8_SDK))
+    set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -headerpad_max_install_names")
+    set(CMAKE_OSX_DEPLOYMENT_TARGET 10.8 CACHE STRING "" FORCE)
+    set(CMAKE_OSX_SYSROOT "/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.8.sdk" CACHE PATH "" FORCE)
+  elseif((${OS_X_VER_MAJOR} VERSION_EQUAL 9) AND (HAVE_10_9_SDK))
```

Comment:
I think its perfectly valid to use 10.8 SDK on say Mac 10.9. But if this works better then I am fine with that.